core/src/test/java/tech/tablesaw/api/TableTest.java

```diff
@@ -547,4 +550,56 @@ private DoubleColumn sum(DoubleColumn... columns) {
         }
         return result;
     }
+
+    private void assertSizeAndMissingAtEnd(int expectedSize, IntColumn col, int expectedMissing) {
+        assertEquals(expectedSize, col.size());        
+        int count = 0;
+        for (int i = col.size() - 1; i >= 0; i--) {
+            if (col.isMissing(i)) {
+                count++;
+            } else {
+                break;
+            }
+        }
+        assertEquals(expectedMissing, count);        
+    }
+
+    @Test
+    public void testPadColumnsWithMissing() {
+        IntColumn col11 = IntColumn.create("col1");
+        IntColumn col12 = IntColumn.create("col2");
+        Table t1 = Table.create("t1", col11, col12);
+        col11.append(1);
+        col12.append(1).append(2);
+        IntColumn col13 = IntColumn.create("col3").append(1).append(2).append(3);
+        t1.padColumnsWithMissing(col13);
+        assertSizeAndMissingAtEnd(3, col11, 2);
+        assertSizeAndMissingAtEnd(3, col12, 1);
+        assertSizeAndMissingAtEnd(3, col13, 0);
+
+        IntColumn col21 = IntColumn.create("col1");
+        IntColumn col22 = IntColumn.create("col2");
+        Table t2 = Table.create("t1", col21, col22);
+        col21.append(1).append(2).append(3);
+        col22.append(1).append(2);
+        IntColumn col23 = IntColumn.create("col3").append(1);
+        t2.padColumnsWithMissing(col23);
+        assertSizeAndMissingAtEnd(3, col21, 0);
+        assertSizeAndMissingAtEnd(3, col22, 1);
+        assertSizeAndMissingAtEnd(3, col23, 2);
+    }
+
+    @Test
+    public void testToStringColumnsWithVaryingSizes() {
+        IntColumn col11 = IntColumn.create("col1");
+        IntColumn col12 = IntColumn.create("col2");
+        Table t1 = Table.create("t1", col11, col12);
+        col11.append(1).append(2);
+        col12.append(1);
+        try {
+            t1.toString();
+        } catch (Exception e) {
+            Assertions.fail("toString shouldn't throw " + e);
```

Comment:
I don't think you need to catch the exception. I believe the test will fail automatically if an exception is thrown

Reply:
Yes, but by including the fail I hoped to make Codacy happy. And it makes it a bit clearer what I the test was about, and it would be reported as a failure, not an error.

Reply:
The `assertNotNull` will do both of those things